src/material/datepicker/calendar.ts

```diff
@@ -82,7 +82,11 @@ export class MatCalendarHeader<D> {
     const minYearOfPage = activeYear - getActiveOffset(
       this._dateAdapter, this.calendar.activeDate, this.calendar.minDate, this.calendar.maxDate);
     const maxYearOfPage = minYearOfPage + yearsPerPage - 1;
-    return `${minYearOfPage} \u2013 ${maxYearOfPage}`;
+    const minYearName =
+      this._dateAdapter.getYearName(this._dateAdapter.createDate(minYearOfPage, 0, 1));
+    const maxYearName =
+      this._dateAdapter.getYearName(this._dateAdapter.createDate(maxYearOfPage, 0, 1));
+    return `${minYearName} \u2013 ${maxYearName}`;
```

Comment:
Technically, proper i18n should allow for formatting this in whatever way makes sense for the locale ("min - max", "max - min", "[min - max]", "min (╯°□°)╯︵ ┻━┻ max", etc). Have we run into this in any other components? and if so do we have an established pattern for dealing with it?

If not, I would think a method on the i18n injectable would be sufficient. `formatRange(start: string, end: string): string`

Reply:
We have something similar in the paginator and [we deal in a very similar way with it](https://github.com/angular/components/blob/master/src/material/paginator/paginator-intl.ts#L53). Doesn't `formatRange` also make some assumptions though? E.g. we aren't specifying what kind of range is being formatted. I can imagine a range of dates being formatted differently compared to a range of numbers.

Reply:
yeah this would go specifically on `MatDatepickerIntl` and we can give it a more specific name that indicates its exact purpose. That was just my placeholder name

Reply:
I took a quick look and I'm not sure that it'll work in `MatDatepickerIntl` because we need access to the specific date adapter, whereas `MatDatepickerIntl` is `providedIn: 'root'`.

Reply:
I was thinking the `MatDatepickerIntl` function could take the dates in as strings, already formatted. I guess its still not ideal, maybe there's some language that writes "2019" different in the context of "2018 - 2019", but its at least as close as we can get to the ideal